src/Testcontainers/Configurations/WaitStrategies/IWaitForContainerOS.cs

```diff
@@ -112,6 +114,25 @@ public interface IWaitForContainerOS
     [PublicAPI]
     IWaitForContainerOS UntilContainerIsHealthy(long failingStreak = 3, Action<IWaitStrategy> waitStrategyModifier = null);
 
+    /// <summary>
+    /// Waits until a connection to the database can be successfully opened.
+    /// </summary>
+    /// <param name="dbProviderFactory">The <see cref="DbProviderFactory" /> used to create the database connection.</param>
+    /// <param name="frequency">
+    /// The retry frequency until either the connection is successfully opened or the <paramref name="timeout" /> is reached.
+    /// Defaults to 1 second if not specified.
+    /// </param>
+    /// <param name="timeout">
+    /// The maximum duration to retry. Once reached, a <see cref="TimeoutException" /> is thrown.
+    /// Defaults to 5 minutes if not specified.
+    /// </param>
+    /// <returns>A configured instance of <see cref="IWaitForContainerOS" />.</returns>
+    /// <remarks>
+    /// This wait strategy must only be applied to containers implementing the <see cref="IDatabaseContainer"/> interface.
+    /// </remarks>
+    [PublicAPI]
+    IWaitForContainerOS UntilDatabaseIsAvailable(DbProviderFactory dbProviderFactory, TimeSpan frequency = default, TimeSpan timeout = default);
```

Comment:
Instead of using `frequency` and `timeout`, we should use the `Action<IWaitStrategy> waitStrategyModifier = null` argument, just like the other wait strategies do. That already gives you the ability to configure the [behaviors](https://github.com/testcontainers/testcontainers-dotnet/blob/2dc9f574a0e05b629bc4dcf9ca6d627aaa233aa7/src/Testcontainers/Configurations/WaitStrategies/IWaitStrategy.cs#L10-L31).

Reply:
I addressed this in f579265e5180a960183027f0db59663a11a17a2b. Indeed it doesn't make sense to reinvent a wait strategy mechanism when there's already one!

I'm just a little sad that by doing so, the timeout exception is degraded.

😀 Before (includes the connection string and the timeout)
```
System.TimeoutException: The database was not available at "Host=127.0.0.1;Port=32897;Database=postgres;Username=postgres;Password=postgres" after waiting for 2 seconds.
```

🙁 After (no useful information at all)
```
System.TimeoutException: The operation has timed out.
```

Reply:
That's something I noticed as well. Maybe we can improve it in a separate PR.